src/components/PhotoLibrary/PhotoGallery.js

```diff
@@ -61,18 +65,60 @@ const PhotoGallery = ( ): Node => {
     if ( !canRequestPhotos ) {
       setCanRequestPhotos( true );
     }
-  } );
+  }, [canRequestPhotos] );
 
-  const navigation = useNavigation( );
+  useEffect( ( ) => {
+    if ( clearSelection ) {
+      // clear selection when opening photo gallery from camera options modal
+      setSelectedPhotos( [] );
+    }
+  }, [clearSelection, setSelectedPhotos, updateSelection] );
+
+  // $FlowIgnore
+  const selectedAlbum = photoOptions.groupName || "All";
+
+  const updatePhotoGallery = useCallback( rerenderFlatList => {
+    setPhotoGallery( {
+      ...photoGallery,
+      // there might be a better way to do this, but adding this key forces the FlatList
+      // to rerender anytime an image is unselected
+      rerenderFlatList
+    } );
+  }, [photoGallery] );
+
+  useEffect( ( ) => {
+    if ( photosFetched ) {
+      if ( photoGallery[selectedAlbum]
+        && photoGallery[selectedAlbum].length === photosFetched.length ) {
+        return;
+      }
+
+      // store photo details in state so it's possible
+      // to select mutiple photos across albums
```

Comment:
Is it even possible to choose photos from a particular album?

Reply:
Yes, if you tap on the camera roll button at the top, you can select other albums.